CMakeLists.txt

```diff
@@ -372,23 +388,79 @@ if(BUILD_TESTING)
            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
-if(UNIX)
-  # Now for some gibbering horrors from beyond the stars...
+# Now for some gibbering horrors from beyond the stars...
+
+if(LIBUV_NO_GNUINSTALLDIRS)
+  set(CMAKE_INSTALL_LIBDIR lib)
+  set(CMAKE_INSTALL_BINDIR bin)
+  set(CMAKE_INSTALL_INCLUDEDIR include)
+  set(CMAKE_INSTALL_DOCDIR share/doc/libuv)
+else()
   include(GNUInstallDirs)
-  foreach(x ${uv_libraries})
-    set(LIBS "${LIBS} -l${x}")
-  endforeach(x)
-  file(STRINGS configure.ac configure_ac REGEX ^AC_INIT)
-  string(REGEX MATCH [0-9]+[.][0-9]+[.][0-9]+ PACKAGE_VERSION "${configure_ac}")
-  set(includedir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
-  set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
-  set(prefix ${CMAKE_INSTALL_PREFIX})
-  configure_file(libuv.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libuv.pc @ONLY)
-
-  install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-  install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DOCDIR})
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libuv.pc
-          DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+endif()
+
+foreach(x ${uv_libraries})
+  set(LIBS "${LIBS} -l${x}")
+endforeach(x)
+file(STRINGS configure.ac configure_ac REGEX ^AC_INIT)
+string(REGEX MATCH [0-9]+[.][0-9]+[.][0-9]+ PACKAGE_VERSION "${configure_ac}")
+set(includedir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
+set(prefix ${CMAKE_INSTALL_PREFIX})
+configure_file(libuv.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libuv.pc @ONLY)
+
+install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DOCDIR})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libuv.pc
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+if(LIBUV_EXPORT_CMAKE)
+  export(TARGETS uv uv_a FILE ${CMAKE_BINARY_DIR}/libuv-targets.cmake)
+  install(
+    EXPORT libuv-targets
+    NAMESPACE libuv::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libuv
+    COMPONENT dev)
+
+  install(TARGETS uv
+          EXPORT libuv-targets
+          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS uv_a
+          EXPORT libuv-targets
+          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+  # Generate the libuv-config.cmake for find_package()
+  file(WRITE ${CMAKE_BINARY_DIR}/libuv-config.cmake
+"# libuv-config.cmake
```

Comment:
I would be OK splitting this into its own file. For the moment, I felt being less intrusive was better.